Add demo code for 2024-july talk (#3)
ballerina-guides/ballerina-tech-talks: 2024-july/README.md

# Ballerina Tech Talk - July 2024

_Topic_: **Beyond CLI Basics: Using Extended Ballerina CLI Tools for Powerful Integrations** \
_Date_: 2024-07-11 \
_Presenter_: [@GayalDassanayake](https://github.com/gayaldassanayake) \

> _Link_: [YouTube](TODO)

Reviewer: We can use the existing YouTube link here. IIRC, that link will not change after making it public.

Author: Updated with c1ece123ac5f3b7661beb48fe4f5a5b93aa8d8c4

> The below code samples use Ballerina Swan Lake Update 9 (2201.9.2)

## Demonstration 1: `bal tool` Command

> In this section we go over the `bal tool` command functionality shown in the first demonstration of the tech talk.

Reviewer 2: ```suggestion
In this section, we review the `bal tool` command functionality shown in the first demonstration of the tech talk.
```

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73

`bal tool` has a set of sub-commands that allow managing the tool life cycle.

### `bal tool search`

> `bal tool search` command is used to search for tool information in the Ballerina Central.

Reviewer 2: Link to ballerina central

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73

> `bal tool search` command is used to search for tool information in the Ballerina Central.

Reviewer 2: ```suggestion
The `bal tool search` command is used to search for tool information in the Ballerina Central.
```

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73

> Run `bal tool search` on the `health` tool. The `health` tool generates Ballerina code for developing healthcare integrations.

Reviewer 2: ```suggestion
Run `bal tool search` to search for the `health` tool. The `health` tool generates Ballerina code for developing healthcare integrations.
```

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73

```
$ bal tool search health

Ballerina Central
=================

|ID                 |PACKAGE           |DESCRIPTION                               |DATE            |VERSION       |
|-------------------|------------------|------------------------------------------|----------------|--------------|
|health             |ballerinax/health |This project contains an extension imp... |2024-05-07-Tue  |2.1.2         |

1 tools found.
```

The latest version of the `health` tool is `2.1.2`.

### `bal tool pull`

Pull the `health` tool to the local machine.
```
$ bal tool pull health
ballerinax/health:2.1.2 [central.ballerina.io ->/Users/gayaldassanayake/.ballerina/repositories/central.ballerina.io/bala/ballerinax/health/2.1.2]  100% [=====================] 93313/93313 KB (0:00:08 / 0:00:00)
ballerinax/health:2.1.2 pulled from central successfully

tool 'health:2.1.2' pulled successfully.
tool 'health:2.1.2' successfully set as the active version.
```

> Alternatively we can specify a version to pull a specific version of the tool.

Reviewer 2: ```suggestion
Alternatively, we can specify a version to pull a specific version of the tool.
```

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73

```
$ bal tool pull health:2.1.1
ballerinax/health:2.1.1 [central.ballerina.io ->/Users/gayaldassanayake/.ballerina/repositories/central.ballerina.io/bala/ballerinax/health/2.1.1]  100% [=====================] 93539/93539 KB (0:00:13 / 0:00:00)
ballerinax/health:2.1.1 pulled from central successfully

tool 'health:2.1.1' pulled successfully.
tool 'health:2.1.1' successfully set as the active version.
```

### `bal tool list`

> `bal tool list` command lists the tools that are installed in the local machine.

Reviewer 2: ```suggestion
The `bal tool list` command lists the tools that are installed in the local machine.
```

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73

```
$ bal tool list
|TOOL ID               |VERSION         |REPO       |
|----------------------|----------------|-----------|
|health                |  2.1.2         |central    |
|health                |* 2.1.1         |central    |

2 tools found.
```

### `bal help`

To make sure that the tool is installed, type `bal help`.

``` bash
$ bal help
NAME
       The build system and package manager of Ballerina

SYNOPSIS
       bal <command> [args]
       bal [OPTIONS]

OPTIONS
       # removed for brevity

COMMANDS
        The available subcommands are:

   # removed for brevity

   Tool Commands:
        health          Ballerina Health Artifact Generator Tool
    
   # removed for brevity

Use 'bal help <command>' for more information on a specific command.
```

We can use `bal help health` to get more information about the `health` tool.

### `bal tool use`

`bal tool use` command is used to set the active version of the tool.

```
$ bal tool use health:2.1.2
tool 'health:2.1.2' successfully set as the active version.
bal tool use health:2.1.1
$ bal tool use health:2.1.1
tool 'health:2.1.1' successfully set as the active version.
```

### `bal tool remove`

> `bal tool remove` command is used to remove a tool from the local machine.

Reviewer 2: ```suggestion
The `bal tool remove` command is used to remove a tool from the local machine.
```

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73

```
$ bal tool remove health:2.1.2
tool 'health:2.1.2' successfully removed.
```

### `bal tool update`

> `bal tool update` command is used to update the tool to the latest version.

Reviewer 2: ```suggestion
The `bal tool update` command is used to update the tool to the latest version.
```

Author: Fixed with 375d148123acc99d3e59a9a4cc34445b650e6b73
